src/Psl/DateTime/DateTimeConvenienceMethodsTrait.php

```diff
@@ -0,0 +1,473 @@
+     * represents the week number according to ISO-8601 standards, which ranges from 1 to 53. The week numbering
+     * follows the ISO-8601 specification, where a week starts on a Monday and the first week of the year is the
+     * one that contains at least four days of the new year.
+     *
+     * Due to the ISO-8601 week numbering rules, the returned year might not always match the Gregorian year
+     * obtained from `$this->getYear()`. Specifically:
+     *
+     *  - The first few days of January might belong to the last week of the previous year if they fall before
+     *      the first Thursday of January.
+     *
+     *  - Conversely, the last days of December might be part of the first week of the following year if they
+     *      extend beyond the last Thursday of December.
+     *
+     * Examples:
+     *  - For the date 2020-01-01, it returns [2020, 1], indicating the first week of 2020.
+     *  - For the date 2021-01-01, it returns [2020, 53], showing that this day is part of the last week of 2020
+     *      according to ISO-8601.
+     *
+     * @return array{int, int<1, 53>}
+     *
+     * @mutation-free
+     */
+    public function getISOWeekNumber(): array
+    {
+        /** @var int<1, 53> $week */
+        $week = (int)$this->format(pattern: 'w');
+        $year = (int)$this->format(pattern: 'Y');
+
+        return [$year, $week];
+    }
+
+    /**
+     * Gets the weekday of the date.
+     *
+     * @return Weekday The weekday.
+     *
+     * @mutation-free
+     */
+    public function getWeekday(): Weekday
+    {
+        return Weekday::from((int) $this->format(pattern: 'e'));
+    }
+
+    /**
+     * Returns whether the year stored in this DateTime object is a leap year
+     * (has 366 days including February 29).
+     */
+    public function isLeapYear(): bool
+    {
+        return namespace\is_leap_year($this->getYear());
+    }
+
+    /**
+     * Adds the specified years to this date-time object, returning a new instance with the added years.
+     *
+     * @throws Exception\UnderflowException If adding the years results in an arithmetic underflow.
+     * @throws Exception\OverflowException If adding the years results in an arithmetic overflow.
+     *
+     * @mutation-free
+     */
+    public function plusYears(int $years): static
+    {
+        if ($years < 0) {
+            return $this->minusYears(-$years);
+        }
+
+        $current_year = $this->getYear();
+
+        // Check for potential overflow when adding years
+        if ($current_year >= Math\INT64_MAX - $years || $current_year <= Math\INT64_MIN - $years) {
+            throw new Exception\OverflowException("Adding years results in a year that exceeds the representable integer range.");
+        }
+
+        $target_year = $current_year + $years;
+
+        // Handle the day and month for the target year, considering leap years
```

Comment:
Just thinking out loud here : would it make sense to just call plusMonths($years * MONTHS_IN_YEAR) here? 
That way the logic does not need to get repeated in plus/minus months and years

Reply:
done